website/blog/2023-11-05-workspace-folders.md

```diff
@@ -0,0 +1,94 @@
+
+### VSCode
+
+In VSCode workspace folder support is achieved by [multi-root workspaces](https://code.visualstudio.com/docs/editor/multi-root-workspaces). To load multiple projects into a single workspace you can simply open one of the projects and add the other one using `File > Add Folder to Workspace` and then choosing the correct folder.
+
+![add-workspace-folder](https://i.imgur.com/LTYrx9V.gif)
+
+Now you have two projects loaded side by side, so you can easily see both and switch between them. All of the current metals functionality accommodates multiple projects, so you can use metals the same way as you did before. The biggest changes will be visible in the places where information from the whole workspace is collected, like workspace symbol search, test explorer, or metals doctor.
+
+![multi-root-tests](https://i.imgur.com/zWmmsC2.gif)
+
+The target project for a command is usually chosen based on the currently opened file. E.g. if you run `Switch build server` the command it will be executed for the project in focus. If no project is in focus the editor will explicitly ask for which project the command should be executed.
+
+![target-folder](https://i.imgur.com/tV7K822.gif)
+
+Finally, logs can still be found in the `.metals/metals.log` in the root of each project. Note, that for the time being all information is logged to all opened workspace folders, so anything logged for `project1` will also be visible in logs for `project2`.
+
+### nvim-metals (_section written by [ckipp01](https://github.com/ckipp01)_)
+
+When using nvim-metals, you'll start just like you do with any other project.
+Since the idea of a workspace is a bit "artificial" with Neovim, you can really
+just add any new root to have a multi-root workspace. All you'll need to do is
+navigate to a file at the root level of the workspace you'd like to add, and use
+the
+[`vim.lsp.buf.add_workspace_folder()`](https://neovim.io/doc/user/lsp.html#vim.lsp.buf.add_workspace_folder()) function to add the folder containing the file you're in as another root.
+
+![add_workspace_folder](https://i.imgur.com/E8iriR9.gif)
+
+To make this easier, you can also just create a mapping to use.
+
+```lua
+vim.keymap.set("n", "<leader>awf", vim.lsp.buf.add_workspace_folder)
+```
+
+To verify that this has worked correctly you should be able to now see both
+workspaces reflected in your Metals Doctor.
+
+![nvim-metals doctor](https://i.imgur.com/2u48wDK.gif)
+
+You should also see that some commands, like the
+[`vim.lsp.buf.workspace_symbol()`](https://neovim.io/doc/user/lsp.html#vim.lsp.buf.workspace_symbol())
+now show results from all the added workspaces.
+
+![workspace_symbols](https://i.imgur.com/RczJIcp.gif)
+
+## Changes for the clients
+
+Since workspace folders are a part of the LSP for any client implementing this capability the multi-root support should work out of the box, however , there will be a few minor changes to accommodate the new approach.
```

Comment:
```suggestion
Since workspace folders are a part of the LSP for any client implementing this capability the multi-root support should work out of the box, however, there will be a few minor changes needed to accommodate the new approach.
```